File: server/scratch/test_ola.js
const axios = require('axios');
const API_KEY = 'Txtd52X2o49O1UlPk7euny0DUhjd65VGQaMWzBoR';
async function test() {
  const origin = '12.9348,77.6189';
  const destination = '12.9716,77.5946';
  for (let mode of ['driving', 'two_wheeler', 'walking']) {
    try {
      const res = await axios.post(`https://api.olamaps.io/routing/v1/directions?origin=${origin}&destination=${destination}&api_key=${API_KEY}&mode=${mode}`);
      const route = res.data.routes[0];
      const dist = route.legs[0].distance;
      const dur = route.legs[0].duration;
      console.log(`Mode: ${mode}, Distance: ${dist}, Duration: ${dur}`);
    } catch(e) {
      console.log(`Mode ${mode} error: ${e.response?.data?.message || e.message}`);
    }
  }
}
test();

File: scratch/test_ola.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

void main() async {
  final apiKey = 'Txtd52X2o49O1UlPk7euny0DUhjd65VGQaMWzBoR';
  final origin = '12.9715987,77.5945627'; // Bangalore
  final dest = '15.2993,74.1240'; // Goa

  final uri = Uri.parse('https://api.olamaps.io/routing/v1/directions').replace(queryParameters: {
    'origin': origin,
    'destination': dest,
    'api_key': apiKey,
    'mode': 'driving'
  });

  final dirRes = await http.post(uri);
  
  if (dirRes.statusCode == 200) {
    final data = jsonDecode(dirRes.body);
    final routes = data['routes'];
    if (routes != null && routes.isNotEmpty) {
      final leg = routes[0]['legs'][0];
      print('Distance (exact coordinates): ${leg['distance']}');
      print('Duration: ${leg['duration']}');
      
      // Look at the readable distance/duration
      if (leg.containsKey('readable_distance')) {
        print('Readable Distance: ${leg['readable_distance']}');
      }
    }
  } else {
    print('Dir Error: ${dirRes.body}');
  }
}
